aws/resource_aws_waf_regex_pattern_set.go

```diff
@@ -0,0 +1,176 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/waf"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsWafRegexPatternSet() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsWafRegexPatternSetCreate,
+		Read:   resourceAwsWafRegexPatternSetRead,
+		Update: resourceAwsWafRegexPatternSetUpdate,
+		Delete: resourceAwsWafRegexPatternSetDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"regex_pattern_strings": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceAwsWafRegexPatternSetCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).wafconn
+
+	log.Printf("[INFO] Creating WAF Regex Pattern Set: %s", d.Get("name").(string))
+
+	wr := newWafRetryer(conn, "global")
+	out, err := wr.RetryWithToken(func(token *string) (interface{}, error) {
+		params := &waf.CreateRegexPatternSetInput{
+			ChangeToken: token,
+			Name:        aws.String(d.Get("name").(string)),
+		}
+		return conn.CreateRegexPatternSet(params)
+	})
+	if err != nil {
+		return fmt.Errorf("Failed creating WAF Regex Pattern Set: %s", err)
+	}
+	resp := out.(*waf.CreateRegexPatternSetOutput)
+
+	d.SetId(*resp.RegexPatternSet.RegexPatternSetId)
+
+	return resourceAwsWafRegexPatternSetUpdate(d, meta)
+}
+
+func resourceAwsWafRegexPatternSetRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).wafconn
+	log.Printf("[INFO] Reading WAF Regex Pattern Set: %s", d.Get("name").(string))
+	params := &waf.GetRegexPatternSetInput{
+		RegexPatternSetId: aws.String(d.Id()),
+	}
+
+	resp, err := conn.GetRegexPatternSet(params)
+	if err != nil {
+		if isAWSErr(err, "WAFNonexistentItemException", "") {
```

Comment:
I see this is related to an upstream SDK issue you filed 😄  https://github.com/aws/aws-sdk-go/issues/1856

Maybe we should note there's a constant available that doesn't seem to include that prefix: `waf.ErrCodeNonexistentItemException` in case a future traveler might go fixing this?